Modules/main.c

```diff
@@ -1101,17 +1064,43 @@ pymain_import_readline(_PyMain *pymain)
 }
 
 
-static FILE*
-pymain_open_filename(_PyMain *pymain, _PyCoreConfig *config)
+static void
+pymain_run_startup(_PyMain *pymain, _PyCoreConfig *config, PyCompilerFlags *cf)
 {
-    FILE* fp;
+    const char *startup = _PyCoreConfig_GetEnv(config, "PYTHONSTARTUP");
+    if (startup == NULL) {
+        return;
+    }
 
-    fp = _Py_wfopen(pymain->filename, L"r");
+    FILE *fp = _Py_fopen(startup, "r");
+    if (fp == NULL) {
+        int save_errno = errno;
+        PySys_WriteStderr("Could not open PYTHONSTARTUP\n");
+        errno = save_errno;
+
+        PyErr_SetFromErrnoWithFilename(PyExc_OSError,
+                        startup);
+        PyErr_Print();
+        PyErr_Clear();
```

Comment:
There's no need to call `PyErr_Clear()` after `PyErr_Print()`.

Reply:
This code is 11 years old, I didn't touch it. Please open a new issue if you want to fix it.

See commit e69a08ea933e69262b08dc18bb3565892de0ad9b.

Reply:
This has been fixed in 8b58468.